### PR TITLE
Added SQL RAND() function.

### PR DESCRIPTION
**Pull request by dmromanov**
Added an SQL RAND() function to ORM.

Because Sqlite is the only DBMS with RANDOM() function returning values not between 0 <= x < 1, I've added some normalization to make the function consistent.

Seeding is not supported.

Refs #12077

### CODE REVIEW

**Comment by saeideng** [comment]
can we have  `$query->func()->rand($seed)` !?

**Comment by dmromanov** [comment]
@saeideng Well, practically, we can but, I have no idea on how to test it. Different DBMS' use different values for seeding. And Sqlite can not seed at all.

File: src/Database/FunctionsBuilder.php [hunk 1]
```diff
@@ -60,6 +60,16 @@ protected function _literalArgumentFunction($name, $expression, $types = [], $re
         return $this->_build($name, $expression, $types, $return);
     }
 
+    /**
+     * Returns a FunctionExpression representing a call to SQL RAND function.
+     *
+     * @return \Cake\Database\Expression\FunctionExpression
+     */
+    public function rand()
```
Comment (markstory) [inline src/Database/FunctionsBuilder.php]: Should the function name be `random()`?

Comment (dmromanov) [reply src/Database/FunctionsBuilder.php]: I'd stick to the short version to match other functions like "avg" and "sum".

Comment (markstory) [reply src/Database/FunctionsBuilder.php]: Fair enough.

**Comment by markstory** [comment]
Moved to 3.7 as this adds new methods.

File: tests/bootstrap.php [hunk 2]
```diff
@@ -139,6 +139,3 @@
 // does not allow the sessionid to be set after stdout
 // has been written to.
 session_id('cli');
-
-// Fix multiple http/server requests in a single test method.
-$_SERVER['PHP_SELF'] = '/';
```
Comment (dereuromark) [inline tests/bootstrap.php]: Why was this removed?

Comment (saeideng) [reply tests/bootstrap.php]: removed in other PR

Comment (dmromanov) [reply tests/bootstrap.php]: @dereuromark I did not do it. Must be one of thouse side effects of rebasing a branch in Git.

File: src/Database/FunctionsBuilder.php (hunk 1 above)
Comment (dmromanov) [reply src/Database/FunctionsBuilder.php]: `rand()` comes from SQL `RAND()` function as in MySQL and SqlServer.
